src/Microsoft.OData.Edm/Schema/CoreModel/EdmCoreModelPathType.cs

```diff
@@ -0,0 +1,55 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmCoreModelPathType.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// The built-in Edm.AnnotationPath, Edm.PropertyPath, Edm.NavigationPropertyPath abstrat type in the core model.
+    /// </summary>
+    internal sealed class EdmCoreModelPathType : EdmType, IEdmPathType, IEdmCoreModelElement
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmCoreModelPathType"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="pathKind">The path kind.</param>
+        public EdmCoreModelPathType(string name, EdmPathTypeKind pathKind)
+        {
+            Name = name ?? string.Empty;

```

Comment:
>string.Empty; [](start = 27, length = 13)

Should we allow string.Empty?  Should we restrict to AnnotationPath, PropertyPath, and NavigationPropertyPath?

Reply:
I modified to use the enum string.

---
In reply to: [164584303](https://github.com/OData/odata.net/pull/1051#discussion_r164584303) [](ancestors = 164584303)